pkg/workload/schemachange/operation_generator.go

```diff
@@ -2364,16 +2364,12 @@ func (og *operationGenerator) setColumnDefault(ctx context.Context, tx pgx.Tx) (
 			return nil, err
 		}
 		if newTyp == nil {
-			errCode := pgcode.UndefinedObject // Error: type 'IrrelevantType'::<newTypeName> does not exist.
-			// Setting default on generated column short-circuits and returns a syntax
-			// error.
-			if columnForDefault.generated {
-				errCode = pgcode.Syntax
-			}
-			return makeOpStmtForSingleError(OpStmtDDL,
-				fmt.Sprintf(`ALTER TABLE %s ALTER COLUMN %s SET DEFAULT 'IrrelevantValue':::%s`,
-					tableName.String(), columnForDefault.name.String(), newTypeName.SQLString()),
-				errCode), nil
+			stmt := makeOpStmt(OpStmtDDL)
+			stmt.potentialExecErrors.add(pgcode.UndefinedObject)
+			stmt.potentialExecErrors.add(pgcode.Syntax)
```

Comment:
just curious, does postgres also return a syntax error for this case? 

also, it could help to add a comment saying which scenario causes this error code.

Reply:
it does -- here is what postgres does:

```
postgres=# CREATE TABLE t (
    id SERIAL PRIMARY KEY,
    value1 INT,
    value2 INT,
    computed_column INT GENERATED ALWAYS AS (value1 + value2) STORED
);
CREATE TABLE
postgres=# alter table t alter column computed_column set default null;
ERROR:  42601: column "computed_column" of relation "t" is a generated column
LOCATION:  ATExecColumnDefault, tablecmds.c:7644
```

also: fair point, let me add comments

Reply:
https://github.com/cockroachdb/cockroach/blob/c73c32e36fce79ebc424184e3f560bd460c5f2b0/pkg/sql/pgwire/pgcode/plpgsql_codenames.go#L175